integrations/marketo-v2/lib/index.js

```diff
@@ -97,18 +103,48 @@ var Marketo = (module.exports = integration('Marketo V2')
  */
 
 Marketo.prototype.initialize = function() {
+  var munchkinId = this.options.accountId;
+  var marketoHostUrl = this.options.marketoHostUrl;
+  var marketoFormId = parseInt(this.options.marketoFormId, 10);
+
+  var identifySettingsAreInvalid =
+    marketoHostUrl === undefined ||
+    marketoHostUrl === '' ||
+    Number.isNaN(marketoFormId) ||
+    marketoFormId <= 0;
+
+  if (identifySettingsAreInvalid) {
+    console.warn(
+      'Invalid settings for identify method. Please review your Marketo V2 destination settings.'
+    );
+    return;
+  }
+
   var self = this;
   this.load(function() {
-    window.Munchkin.init(self.options.accountId, {
+    window.Munchkin.init(munchkinId, {
       asyncOnly: true
     });
+
     // marketo integration actually loads a marketo snippet
     // and the snippet loads the real marketo, this is required
     // because there's a race between `window.mktoMunchkinFunction = sinon.spy()`
     // and marketo's real javascript which overrides `window.mktoMunchkinFunction`
     // and deletes the spy.
     when(self.loaded, self.ready);
   });
+
+  this.load('forms', { marketoHostUrl: marketoHostUrl }, function() {
+    var marketoForm = document.createElement('form');
```

Comment:
Should we only create the form id `marketoHostUrl` is defined ?

Reply:
There's a validation for that already. The code won't reach this point if `marketoHostUrl` is undefined